Merge release/keyvault-4.1.0 into master

heaths opened the pull request:
This needed a separate merge commit since there were lots of conflicts expected, and re-recordings were needed.

heaths committed:
Add APIs for Key Vault 4.1.0 (#13548)

* Add support for and change latest version to use 7.1

* Add RecoverableDays to properties and import to KeyOperations

* Update public APIs

* Re-record tests for Key Vault 7.1

heaths committed:
Update versions for Key Vault 4.1 release (#14145)

Closes #13015

heaths committed:
Copy older 4.1.0-preview.1 changelogs (#14161)

We released a couple servicing versions after shipping 4.1.0-preview.1, so it made the new 4.1.0 CHANGELOG entries look really lightweight.

heaths committed:
Merge release/keyvault-4.1.0 into master

isra-fel committed:
Update CODEOWNERS (#14197)

Remove @isra-fel and @erich-wang from mangement plane code owners

vipeketi committed:
<QnAMaker> GenerateAnswer new field "StrictFiltersCompoundOperationType" (#13936)

* Metadata Operation Type

* generated

* Update cognitiveservices_data-plane_QnAMaker.txt

folder stucture

* Changelog csproj version changes

moreOver0 committed:
[Cognitive Service Anomaly Detector] update sdk (#14177)

* Anomaly Detector new release

* update version

* update the changelog

* update changelog

* Update release date

* update CHANGELOG.md

* update CHANGELOG.md

* Change release date

Co-authored-by: Alexander Sher <AlexanderSher@users.noreply.github.com>

christothes committed:
add more CosmosThrottleWrappers (#14208)

pakrym committed:
Update Autorest C# - AppConfiguration, Compute, testcommon (#13525)

Co-authored-by: Harvey Chen <30464227+HarveyLink@users.noreply.github.com>
Co-authored-by: Youri <v-lipyo@microsoft.com>

pakrym committed:
Update Autorest C# - storage (#13582)

Co-authored-by: Harvey Chen <30464227+HarveyLink@users.noreply.github.com>

pakrym committed:
Update Autorest C# - eventhub (#13585)

Co-authored-by: Youri <v-lipyo@microsoft.com>
Co-authored-by: Harvey Chen <30464227+HarveyLink@users.noreply.github.com>

timtay-microsoft committed:
Add query client to IoT Hub service client (#14210)

pakrym committed:
Update Autorest C# - resources (#13580)

pakrym committed:
Update Autorest C# - network (#13584)

* Update Autorest C# - network

* Update Network Test Record

Update Network Test Record

* regen

* Export api

Co-authored-by: Harvey Chen <30464227+HarveyLink@users.noreply.github.com>

christothes commented on sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClientOptions.cs:
```diff
@@ -31,9 +31,9 @@ public enum ServiceVersion
 
             /// <summary>
-            /// The Key Vault API version 7.1-preview.
+            /// They Key Vault API version 7.1.
```
They => The

christothes replied:
looks like this is in a few other places as well.

heaths replied:
Oops. Well, that shipped. They're all fixed moving forward. 🤦‍♂️

christothes reviewed (approved):
Looks good, with the exception of the They/The typo.

heaths committed:
Resolve PR feedback

seanmcc-msft committed:
Removed IEquatable interface from TaggedBlobItem (#14212)

jaschrep-msft committed:
Use ConcurrentBag when testing progress reports (#14213)

* Use ConcurrentBag when testing progress reports

* Fixed check in progress tests.

Progress tests now check max value of reports, instead of last, now that
their grouping is unordered.

* Fix 0-element list race condition.

Sometimes Progress<T> fails to report anything before we start checking
the collection of its outputs, which causes IEnumerable<T>::Max() to
throw. Added safety for this check.

vinagesh committed:
feat(samples): Added bulk module and query samples (#14215)

seanmcc-msft committed:
Preparing Storage for release (#14218)

heaths committed:
Merge branch 'merge-keyvault-4.1.0'
